packages/tupaia-web/src/features/Map/useDefaultMapOverlay.ts

```diff
@@ -0,0 +1,45 @@
+/*
+ * Tupaia
+ *  Copyright (c) 2017 - 2023 Beyond Essential Systems Pty Ltd
+ */
+
+import { useEffect } from 'react';
+import { useSearchParams } from 'react-router-dom';
+import { useProject } from '../../api/queries';
+import {
+  DEFAULT_MAP_OVERLAY_ID,
+  DEFAULT_PERIOD_PARAM_STRING,
+  URL_SEARCH_PARAMS,
+} from '../../constants';
+import { MapOverlayGroup, ProjectCode, EntityCode } from '../../types';
+
+// When the map overlay groups change, update the default map overlay
+export const useDefaultMapOverlay = (
```

Comment:
I ended up refactoring this a little to try and make it clearer and moved it to the map component so it's easier to find.